modules/services/gpg-agent.nix

```diff
@@ -29,10 +29,44 @@ in
         '';
       };
 
+      defaultCacheTtlSsh = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Set the time a cache entry used for SSH keys is valid to the given number of seconds.
+        '';
+      };
+
       enableSshSupport = mkOption {
         type = types.bool;
         default = false;
-        description = "Whether to use the GnuPG key agent for SSH keys.";
+        description = ''
+          Whether to use the GnuPG key agent for SSH keys.
+        '';
+      };
+
+      noGrab = mkOption {
```

Comment:
I'd suggest to instead have `grabKeyboardAndMouse = mkOption { … }` defaulting to `true`. When possible I'd prefer avoiding negated options for pedagogical reasons. Can have `optional (!cfg.grabKeyboardAndMouse) "no-grab"` below.